mesonbuild/environment.py

```diff
@@ -1186,10 +1194,11 @@ def detect_fortran_compiler(self, for_machine: MachineChoice):
                         exe_wrap, full_version=full_version)
 
                 if 'PGI Compilers' in out:
-                    linker = PGIDynamicLinker(
-                        compiler, for_machine, 'pgi',
-                        PGIFortranCompiler.LINKER_PREFIX, version=version)
-                    return PGIFortranCompiler(
+                    cls = PGIFortranCompiler
+                    self.coredata.add_lang_args(cls.language, cls, for_machine, self)
+                    linker = PGIDynamicLinker(compiler, for_machine, 'pgi',
+                        cls.LINKER_PREFIX, [], version=version)
```

Comment:
__[Flake8]__ continuation line under-indented for visual indent ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6407#issue-5022356))

<details>

| Rule |
| - |
| E128 |

</details>
<sub>posted by Sider</sub>